doc/RPC.md

```diff
@@ -0,0 +1,140 @@
+RPC
+===
+
+Sia peers communicate with each other via Remote Procedure Calls. An RPC consists of a unique ID and a pair of functions, one on the calling end and one on the receiving end. After the ID is written/received, both peers hand the connection off to their respective functions. Typically, the calling end writes an object to the receiver and then reads a response.
+
+RPC IDs are always 8 bytes and contain a human-readable name for the RPC. If the name is shorter than 8 bytes, the remainder is padded with zeros. If the name is longer than 8 bytes, it is truncated.
+
+### Call Listing
+
+Unless otherwise, specified, these calls follow a request/response pattern and use the [encoding](https://github.com/NebulousLabs/Sia/pull/Encoding.md) package to serialize data.
+
+**All data received via RPC should be considered untrusted and potentially malicious.**
+
+#### ShareNodes
+
+ShareNodes requests peer addresses from a peer.
+The gateway calls this RPC regularly to update its list of potential peers.
+
+ID: `"ShareNod"`
+
+Request: None
+
+Response:
+
+```go
+[]modules.NetAddress
+```
+
+Recommendations:
+
++ Requesting peers should limit the request to 3000 bytes.
++ Responding peers should send no more than 10 peers, and should not send peers that are unlikely to be reachable.
+
+#### SendBlocks
+
+SendBlocks requests blocks from a peer. The blocks are added to the requesting peer's blockchain, and optionally rebroadcast to other peers. Unlike most RPCs, the SendBlocks call is a loop of requests and responses that continues until the responding peer has no more blocks to send.
+
+ID: `"SendBloc"`
+
+Request:
+
+```go
+// Exponentially-spaced IDs of most-recently-seen blocks,
+// ordered from most recent to least recent.
+// Less than 32 elements may be present, but the last element
+// (index 31) is always the ID of the genesis block.
+[32]types.BlockID
+```
+
+Response:
+
+```go
+struct {
+   // sequential list of blocks, beginning with the first
+   // block in the main chain not seen by the requesting peer.
+   blocks []types.Block
+   // true if the responding peer can send more blocks
+   more bool
+}
+```
+
+Recommendations:
+
++ Requesting peers should limit the request to 20MB.
++ Responding peers should identify the most recent BlockID that is in their blockchain, and send up to 10 blocks following that block.
++ Responding peers should set `more = true` if they have not sent the most recent block in their chain.
+
+#### RelayBlock
+
+RelayBlock receives a block from a peer, which sent it as a broadcast.
+If the received block is an orphan,
+it will try to find its parents, using the `SendBlocks` RPC.
+Each receiving peer broadcasts the received block to all its peers as well.
```

Comment:
Stuff about rebroadcasting should be moved to the `Recommendations` section. I think the descriptions should be written from the perspective of the peer that initiated the call. So this would become:

>RelayBlock sends a block to a peer, with the expectation that the peer will relay the block to its own peers.
>This RPC is deprecated as of v0.5.1.

Actually, @DavidVorick, should we just strip out the `RelayBlock` RPC entirely? It's been over a year since we deprecated it.